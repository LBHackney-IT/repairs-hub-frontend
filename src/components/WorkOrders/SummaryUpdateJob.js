import PropTypes from 'prop-types'
import { useForm } from 'react-hook-form'
import { PrimarySubmitButton } from '../Form'

const SummaryUpdateJob = ({
  reference,
  onJobSubmit,
  task,
  sorCodesCollection,
  changeStep,
}) => {
  const { handleSubmit } = useForm({})

  const tableData = (data) => {
    return data.map((t, index) => (
      <tr className={`task-${index} govuk-table__row`} key={index}>
        <th
          scope="row"
          className={` sor-code-${index} govuk-table__header`}
          key={t.code}
        >
          {t.code}
        </th>
        <td className={`quantity-${index} govuk-table__cell`} key={t.quantity}>
          {t.quantity}
        </td>
        <td className={`edit-${index} govuk-table__cell`} key={`edit-${index}`}>
          <a onClick={changeStep} href="#">
            Edit
          </a>
        </td>
      </tr>
    ))
  }

  return (
    <div>
      <h1 className="govuk-heading-l">Update work order: {reference}</h1>
      <form
        role="form"
        id="repair-request-form"
        onSubmit={handleSubmit(onJobSubmit)}
      >
        <p className="govuk-heading-s">Summary of updates to work order</p>
        <p className="govuk-heading-s">Tasks and SORs</p>
        <table className="govuk-table">
          <thead className="govuk-table__head">
            <tr className="govuk-table__row">
              <th scope="col" className="govuk-table__header">
                SOR code
              </th>
              <th scope="col" className="govuk-table__header">
                Quantity
              </th>
              <th scope="col" className="govuk-table__header">
                {''}
              </th>
            </tr>
          </thead>
          <tbody className="govuk-table__body">
            {tableData(task)}
            {sorCodesCollection ? tableData(sorCodesCollection) : ''}
          </tbody>
        </table>
        <PrimarySubmitButton label="Confirm and close" />
      </form>
    </div>
  )
}

SummaryUpdateJob.propTypes = {
  reference: PropTypes.string.isRequired,
  onJobSubmit: PropTypes.func.isRequired,
  sorCodesCollection: PropTypes.array,
  changeStep: PropTypes.func.isRequired,
}

export default SummaryUpdateJob
