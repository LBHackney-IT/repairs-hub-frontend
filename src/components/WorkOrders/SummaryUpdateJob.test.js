import { render } from '@testing-library/react'
import SummaryUpdateJob from './SummaryUpdateJob'

describe('SummaryUpdateJob component', () => {
  const props = {
    reference: '10000012',
    originalTasks: [
      {
        code: 'DES5R006',
        quantity: 2,
        original: true,
        originalQuantity: 2,
        cost: '10.5',
      },
    ],
    tasks: [
      {
        code: 'DES5R006',
        quantity: 3,
        original: true,
        originalQuantity: 2,
        cost: '10.5',
      },
      {
        code: 'DES5R005',
        quantity: 1,
        original: false,
        originalQuantity: null,
        cost: '14.5',
      },
    ],
    addedTasks: [
      {
        code: 'DES5R004',
        quantity: 2,
        original: false,
        originalQuantity: null,
        cost: '22.5',
      },
    ],
    onJobSubmit: jest.fn(),
    changeStep: jest.fn(),
    calculateTotalCost: jest.fn(),
  }

  it('should render properly', () => {
    const { asFragment } = render(
      <SummaryUpdateJob
        reference={props.reference}
        onJobSubmit={props.onJobSubmit}
        originalTasks={props.originalTasks}
        tasks={props.tasks}
        addedTasks={props.addedTasks}
        changeStep={props.changeStep}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
