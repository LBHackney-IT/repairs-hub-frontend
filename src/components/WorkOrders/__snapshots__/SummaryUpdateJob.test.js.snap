// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryUpdateJob component should render properly 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="govuk-heading-l"
    >
      Update work order: 10000012
    </h1>
    <form
      id="repair-request-form"
      role="form"
    >
      <p
        class="govuk-heading-s"
      >
        Summary of updates to work order
      </p>
      <p
        class="govuk-heading-s"
      >
        Original Tasks and SORs
      </p>
      <table
        class="govuk-table original-tasks-table"
      >
        <thead
          class="govuk-table__head"
        >
          <tr
            class="govuk-table__row govuk-body"
          >
            <th
              class="govuk-table__header govuk-!-width-one-half"
              scope="col"
            >
              SOR code
            </th>
            <th
              class="govuk-table__header govuk-!-width-one-quarter"
              scope="col"
            >
              Quantity
            </th>
            <th
              class="govuk-table__header govuk-!-width-one-quarter"
              scope="col"
            >
              Cost (unit)
            </th>
            <th
              class="govuk-table__header"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          class="govuk-table__body"
        >
          <tr
            class="govuk-table__row"
            id="original-task-0"
          >
            <th
              class="govuk-table__header"
              scope="row"
            >
              DES5R006
            </th>
            <td
              class="govuk-table__cell"
            >
              2
            </td>
            <td
              class="govuk-table__cell"
            >
              £10.5
            </td>
          </tr>
        </tbody>
      </table>
      <p
        class="govuk-heading-s"
      >
        Updated Tasks and SORs
      </p>
      <table
        class="govuk-table updated-tasks-table"
      >
        <thead
          class="govuk-table__head"
        >
          <tr
            class="govuk-table__row govuk-body"
          >
            <th
              class="govuk-table__header govuk-!-width-one-half"
              scope="col"
            >
              SOR code
            </th>
            <th
              class="govuk-table__header govuk-!-width-one-quarter"
              scope="col"
            >
              Quantity
            </th>
            <th
              class="govuk-table__header govuk-!-width-one-quarter"
              scope="col"
            >
              Cost (unit)
            </th>
            <th
              class="govuk-table__header"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          class="govuk-table__body"
        >
          <tr
            class="govuk-table__row"
            id="existing-task-0"
          >
            <th
              class="govuk-table__header"
              scope="row"
            >
              DES5R006
            </th>
            <td
              class="govuk-table__cell"
            >
              3
            </td>
            <td
              class="govuk-table__cell"
            >
              £10.5
            </td>
            <td
              class="govuk-table__cell"
            >
              <a
                href="#"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="govuk-table__row"
            id="existing-task-1"
          >
            <th
              class="govuk-table__header"
              scope="row"
            >
              DES5R005
            </th>
            <td
              class="govuk-table__cell"
            >
              1
            </td>
            <td
              class="govuk-table__cell"
            >
              £14.5
            </td>
            <td
              class="govuk-table__cell"
            >
              <a
                href="#"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="govuk-table__row"
            id="added-task-0"
          >
            <th
              class="govuk-table__header"
              scope="row"
            >
              DES5R004
            </th>
            <td
              class="govuk-table__cell"
            >
              2
            </td>
            <td
              class="govuk-table__cell"
            >
              £22.5
            </td>
            <td
              class="govuk-table__cell"
            >
              <a
                href="#"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="govuk-table__row"
            id="original-cost"
          >
            <th
              scope="row"
            />
            <td
              class="govuk-!-padding-top-3"
            >
              <strong>
                Original cost
              </strong>
            </td>
            <td
              class="govuk-!-padding-top-3"
            >
              £21.00
            </td>
            <td />
          </tr>
          <tr
            class="govuk-table__row"
            id="variation-cost"
          >
            <th
              scope="row"
            />
            <td
              class="govuk-!-padding-top-3"
            >
              <strong>
                Variation cost
              </strong>
            </td>
            <td
              class="govuk-!-padding-top-3"
            >
              £70.00
            </td>
            <td />
          </tr>
          <tr
            class="govuk-table__row"
            id="total-cost"
          >
            <th
              scope="row"
            />
            <td
              class="govuk-!-padding-top-3 border-top-black"
            >
              <strong>
                Total cost
              </strong>
            </td>
            <td
              class="govuk-!-padding-top-3 border-top-black"
            >
              £91.00
            </td>
            <td />
          </tr>
        </tbody>
      </table>
      <div
        class="govuk-form-group"
      >
        <button
          class="govuk-button"
          data-module="govuk-button"
          type="submit"
        >
          Confirm and close
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
